lib/req/steps.ex

```diff
@@ -1041,7 +1041,8 @@ defmodule Req.Steps do
     Logger.debug(["follow_redirects: redirecting to ", location])
 
     location_trusted = Map.get(request.options, :location_trusted)
-    location_url = URI.parse(location)
+
+    location_url = location |> URI.parse() |> inherit_from_url(request.url)
```

Comment:
I wonder if it could have been simply:

```elixir
location_url = URI.merge(request.url, URI.parse(location))
```

with limited testing it seems to do what we want:

```elixir
iex> test = fn a, b -> URI.merge(URI.parse(a), URI.parse(b)) |> URI.to_string() end

iex> test.("https://foo/foo?foo=1", "https://bar/bar?bar=1")
"https://bar/bar?bar=1"

iex> test.("https://foo/foo?foo=1", "//bar/bar?bar=1")
"https://bar/bar?bar=1"

iex> test.("https://foo/foo?foo=1", "/bar?bar=1")
"https://foo/bar?bar=1"
```

if that sounds good, a PR would be appreciated as always. :D